### ISSUE TITLE
RuntimeError when importing on Mac OS X

### ISSUE THREAD

**Issue by rayosborn** [lmfit/lmfit-py#470]
I am running Python 3.6.3 using a conda installation on Mac OS X v0.12.6. Importing lmfit v0.9.9 raises a RuntimeError when initializing the Matplotlib macosx backend. Although I could rerun python as `pythonw` to fix the error, I am not interested in using the plotting built into lmfit. The same issue forces me to bury lmfit imports within the runtime code in NeXpy (http://nexpy.github.io/nexpy/), which uses lmfit but has its own Matplotlib PyQt backend. It would be helpful to me if the lmfit import doesn't automatically load the Matplotlib backend until it is needed.
```
$ python
Python 3.6.3 | packaged by conda-forge | (default, Nov  4 2017, 10:13:32) 
[GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import lmfit
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rosborn/anaconda/envs/py36/lib/python3.6/site-packages/lmfit/__init__.py", line 48, in <module>
    from .model import Model, CompositeModel
  File "/Users/rosborn/anaconda/envs/py36/lib/python3.6/site-packages/lmfit/model.py", line 41, in <module>
    from matplotlib import pyplot as plt
  File "/Users/rosborn/anaconda/envs/py36/lib/python3.6/site-packages/matplotlib/pyplot.py", line 115, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/Users/rosborn/anaconda/envs/py36/lib/python3.6/site-packages/matplotlib/backends/__init__.py", line 62, in pylab_setup
    [backend_name], 0)
  File "/Users/rosborn/anaconda/envs/py36/lib/python3.6/site-packages/matplotlib/backends/backend_macosx.py", line 17, in <module>
    from matplotlib.backends import _macosx
RuntimeError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends. If you are using (Ana)Conda please install python.app and replace the use of 'python' with 'pythonw'. See 'Working with Matplotlib on OSX' in the Matplotlib FAQ for more information.
```

**Comment by newville** [comment]
@rayosborn  Sorry for the trouble, and I'm sort of surprised I haven't seen this error myself. I struggle with the framework problem with anaconda python on Macs all the time with wxPython apps.

But for sure, lmfit should not require matplotlib or even a working matplotlib.   Right now we're doing in `model.py`:

```python
try:
    from matplotlib import pyplot as plt
    _HAS_MATPLOTLIB = True
except ImportError:
    _HAS_MATPLOTLIB = False
```
which is missing the `RuntimeError` that matplotlib is raising.  
 
I think we should just change this to a bare `except`.   That is, we don't really care why the import of matplotlib might fail, but if it does fail we want to know and carry on without it.

**Comment by rayosborn** [comment]
Thanks for the quick response. I was surprised the issue hadn't been reported already and initially assumed it was something to do with the way I initialize the backend. However, I then noticed it occurs in a regular Python shell.

**Comment by reneeotten** [comment]
just adding the ```RuntimeError``` here should be sufficient as I don't really see another way it could fail... In my opinion that would be better than a bare except, but in this specific case it might be okay to do so.

**Comment by newville** [comment]
@reneeotten I think perhaps `except StandardError` is what we want. That includes almost everything except KeyboardInterrupt, the iteration sentinels,  and SystemExit.

**Comment by reneeotten** [comment]
not a bad idea and that would work for Python 2 - however, the ```StandardError``` class is removed in Python 3 ;)

**Comment by reneeotten** [comment]
@newville looking a bit more at the documentation we could use ```Exception``` instead, that pretty much includes the same stuff as you mentioned... However, I still think that just the ```RuntimeError``` should suffice, but it's your call.

**Comment by newville** [comment]
@reneeotten Good point -- apparently I was reading the old docs!  It seems like `except Exception` is the preferred choice.  Honestly, I don't know why that is significantly different from a bare `except`, except that PEP8 complains about one but not the other.

We really do mean "try to use matplotlib, but if **for any reason** you cannot, carry on without it".  That seems like a bare except to me.

**Comment by reneeotten** [comment]
closed by PR #472, commit 1f22c6f087521a8ec36dd7954bf41b890b07a029
